hermes/probe/create/create.go

```diff
@@ -0,0 +1,186 @@
+// limitations under the License.
+//
+// Author: Alicja Kwiecinska, GitHub: alicjakwie
+//
+// create contains all of the logic necessary to create the files with randomized contents in GCS
+package create
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"github.com/googleinterns/step224-2020/hermes/probe"
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+
+	pb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+const (
+	FileNamePrefixSize      = 9    // len("Hermes_ID")
+	begin                   = 1    // begin is the ID of the first HermesFile.
+	end                     = 50   // end is the ID of the last HermesFile.
+	maxFileSizeBytes        = 1000 // maxFileSizeBytes is the maximum allowed file size in bytes.
+	hermesAPILatencySeconds = "hermes_api_latency_s"
+)
+
+type randomFile struct {
+	id   int32 // id is a positive integer in the range [begin,end]
+	size int   // size is the file size in bytes.
+}
+
+type randomFileReader struct {
+	size int // size in bytes
+	i    int // currently reading this byte
+	rand *rand.Rand
+}
+
+// Read implements the standard Read interface, it reads data from a randomFileReader.
+// Argument:
+//      buf: a byte slice serves as a buffer
+// Returns:
+//	n:  the size of the buffer buf following a Read operation
+//	err: error it should be nil when not done reading and io.EOF once the whole file contents have been read
+func (r *randomFileReader) Read(buf []byte) (n int, err error) {
+	// check whether Read is done
+	if r.i >= r.size {
+		return 0, io.EOF
+	}
+	b := buf
+	if len(buf) > (r.size - r.i) {
+		// if the length of buffer is greater than the number of bytes left to read make the size of the buffer match the number of bytes left to read
+		b = buf[:(r.size - r.i)]
+	}
+	n, err = r.rand.Read(b) // n is now the length  of the buffer
+	if err != nil {
+		return n, err // in this case n = 0
+	}
+	r.i += n
+	return n, err
+}
+
+// NewReader creates a new randomFileReader and returns a pointer to it
+// Returns:
+//	*randomFileReader: a Reader that supports randomFiles. This Reader contains information about the file size in bytes as well as a pseudo-random bytes generator
+// Warning: The randomFileReader produced by NewReader is not thread safe.
+func (f *randomFile) NewReader() *randomFileReader {
+	//  id will serve as a Seed and i - index of the currently read byte  will be set to 0 automatically in the returned reader
+	return &randomFileReader{
+		size: f.size,
+		rand: rand.New(rand.NewSource(int64(f.id))), // rand.NewSource() requires an int64 as an argument
+	}
+}
+
+func newRandomFile(fileID int32, fileSize int) (*randomFile, error) {
+	if fileID < begin || fileID > end {
+		return nil, fmt.Errorf("the file id provided %d wasn't in the required range [%d, %d]", fileID, begin, end)
```

Comment:
Nit: "the file ID".

Reply:
I had ID capitalised but randomFile has a field: id (as we don't export it) so I am wondering which one to use in the comment?

Reply:
If you are referring to the `fileID` argument, then use "fileID" instead of "file id" or "file ID".
You can make the error message more succinct:
`fmt.Errorf("invalid argument: fileID = %d; want %d <= fileID <= %d", ...)`

Reply:
changed!